src/components/popper-table-modal.tsx

```diff
@@ -107,39 +111,49 @@ export const PopperTableModal = observer(
             />
           </div>
 
-          <InputSearch
-            value={searchValue}
-            placeholder={searchInputPlaceholder}
-            onChange={e => onChange && onChange(e.target.value)}
-          />
-          {viewType && setViewType && (
-            <ViewTypeTable setViewType={setViewType} viewType={viewType} />
+          {!isSettingTable && (
```

Comment:
popper-table-modal is a generic component that renders different stuff inside. All that it should know, is to display search bar or not; and display list of items or not. So first you need to receive a prop: isShowSearchBar (true by default). Second, You can move `<ViewTypeTable` from popper-table to columns-list.tsx. And you should operate hiding and showing these items from columns-list.store.ts, popper-table doesn't have to know anything about it.